src/scripts/main.js

```diff
@@ -1,3 +1,383 @@
 'use strict';
 
-// write your code here
+const FIELD_SIZE = 4;
+const AMOUNT_CELLS = FIELD_SIZE * FIELD_SIZE;
+let x = 0;
+let y = 0;
+let number = 0;
+let notMove;
+
+const tbody = document.querySelector('tbody');
+const rows = tbody.children;
+const header = document.querySelector('.game-header');
+const button = header.querySelector('button');
+const scoreValue = document.querySelector('.game-score');
+
+button.addEventListener('click', start);
+
+document.addEventListener('keydown', e => {
+  switch (e.key) {
+    case 'ArrowDown':
+      down();
+      startToRestart('restart');
+
+      if (notMove) {
+        newCell();
+      }
+      break;
+
+    case 'ArrowUp':
+      up();
+      startToRestart('restart');
+
+      if (notMove) {
+        newCell();
+      }
+      break;
+
+    case 'ArrowRight':
+      right();
+      startToRestart('restart');
+
+      if (notMove) {
+        newCell();
+      }
+      break;
+
+    case 'ArrowLeft':
+      left();
+      startToRestart('restart');
+
+      if (notMove) {
+        newCell();
+      }
+      break;
+  }
+
+  const freeCell = some(rows, '');
+
+  if (!notMove && !freeCell) {
+    let count = 0;
+
+    for (let i = 0; i < FIELD_SIZE - 1; i++) {
+      for (let j = 0; j < FIELD_SIZE - 1; j++) {
+        const currentCell = rows[i].children[j];
+        const downCell = rows[i + 1].children[j];
+        const nextCell = rows[i].children[j + 1];
+
+        if (currentCell.innerText === downCell.innerText
+          || currentCell.innerText === nextCell.innerText) {
+          count++;
+        }
+      }
+    }
+
+    if (!count) {
+      showMessage('lose');
+    }
+  }
+
+  const win = some(rows, '2048');
+
+  if (win) {
+    showMessage('win');
+  }
+});
+
+function clearCell(cell) {
+  cell.innerText = '';
+  cell.className = 'field-cell';
+}
+
+function start() {
+  reset();
+  startGame();
+  showMessage('not');
+}
+
+function reset() {
+  [...rows]
+    .map(row => [...row.children]
+      .map(cell => {
+        cell.innerText = '';
+        cell.className = 'field-cell';
+      }));
```

Comment:
what is the purpose of this action? It doesn't affect anything

Reply:
here I clear all cells for reset